server/controllers/FollowController.js

```diff
@@ -0,0 +1,172 @@
+// import passport from 'passport';
+import models from '../database/models';
+import profileHelper from '../utils/helpers/profileHelper';
+
+const { User, Follow } = models;
+
+/**
+ *
+ * The FollowController contains static methods that are used as
+ * controllers to handle the different Follow routes.
+ *  @class FollowController
+ *  @returns {undefined} this is a class thus does not return anything
+ */
+class FollowController {
+  /**
+   * add a user to the follower list of an user
+   * @param {Object} req the request body
+   * @param {Object} res the response body
+   * @param {function} next a call to the next function
+   * @returns {Object} the response body
+   */
+  static follow(req, res, next) {
+    const { id: followerId } = req.body.decoded;
+    const followingId = parseInt(req.params.userId, 10);
+
+    // // Ensure a user cannot unfollow him/her self
+    if (followingId === followerId) {
+      return res.status(422).json({
+        errors: { message: 'You cannot follow yourself' },
+      });
+    }
+    // Get the user being followed
+    User.findById(followingId)
+      .then((user) => {
+      // Return error message if user is not found
+        if (!user) {
+          return res.status(404).json({
+            errors: { message: 'User not found' },
+          });
+        }
+        Follow.findOrCreate({ where: { followerId, followingId } })
+          .spread((userFollower, created) => {
+          // No action is performed if you already following an user
+            if (!created) {
+              return res.status(422).json({
+                errors: { message: 'You are already following this User' },
+              });
+            }
+            // Return user's profile
+            return res.status(201).json({
+              profile: profileHelper.userProfile(user, true),
+            });
+          });
+      }).catch(next);
+    return null;
+  }
+
+  /**
+   * remove a user from the follower list of an user
+   * @param {Object} req the request body
+   * @param {Object} res the response body
+   * @param {function} next a call to the next function
+   * @returns {Object} the response body
+   */
+  static unfollow(req, res, next) {
+    const { userId: followingId } = req.params;
+    const { id: followerId } = req.body.decoded;
+
+    // Get the user being followed
+    User.findById(followingId)
+      .then((user) => {
+      // Return error message if user is not found
+        if (!user) {
+          return res.status(404).json({
+            errors: { message: 'User not found' },
+          });
+        }
+        // Remove user-follower relationship
+        user.removeFollower(followerId)
+          .then((follow) => {
+            if (!follow) {
+              return res.status(422).json({
```

Comment:
why do you use 422(Unprocessable Entity) when the server understands the request? I think `404 or 400`  makes  more sense